marianas_process_loop_umi_1.8.0/marianas_process_loop_umi.cwl

```diff
@@ -0,0 +1,101 @@
+class: CommandLineTool
+cwlVersion: v1.0
+$namespaces:
+  sbg: 'https://www.sevenbridges.com/'
+id: marianas_process_loop_umi_cwl
+baseCommand:
+  - java
+inputs:
+  - id: fastq1
+    type: File?
+    inputBinding:
+      position: 1
+  - id: fastq2
+    type: File?
+  - id: sample_sheet
+    type: File?
+  - id: umi_length
+    type: int?
+    inputBinding:
+      position: 2
+  - id: add_rg_SM
+    type: string?
+outputs:
+  - id: processed_fastq_1
+    type: File?
+    outputBinding:
+      glob: $('*/' + inputs.fastq1.basename)
+      outputEval: |-
+        ${
+            self[0].basename = inputs.add_rg_SM + '_R1.fastq.gz';
+            return self[0]
+        }
+  - id: processed_fastq_2
+    type: File?
+    outputBinding:
+      glob: '$(''*/'' + inputs.fastq2.basename.replace(''_R1_'', ''_R2_''))'
+      outputEval: |-
+        ${
+            self[0].basename = inputs.add_rg_SM + '_R2.fastq.gz';
+            return self[0]
+        }
+  - id: clipping_info
+    type: File?
+    outputBinding:
+      glob: '*/info.txt'
+      outputEval: |-
+        ${
+            self[0].basename = inputs.add_rg_SM + '_info.txt';
+            return self[0]
+        }
+  - id: clipping_dir
+    type: Directory?
+    outputBinding:
+      glob: '*/'
+      outputEval: |-
+        ${
+            self[0].basename = inputs.add_rg_SM + '_umi_clipping_results';
+            return self[0]
+        }
+label: marianas_process_loop_umi.cwl
+arguments:
+  - position: 0
+    prefix: ''
+    separate: false
+    valueFrom: '-server'
+  - position: 0
+    prefix: ''
+    separate: false
+    valueFrom: '-Xms$(parseInt(runtime.ram)/1000 - 2)g'
+  - position: 0
+    prefix: ''
+    separate: false
+    valueFrom: '-Xmx$(parseInt(runtime.ram)/1000 - 2)g'
+  - position: 0
+    prefix: ''
+    separate: false
+    valueFrom: '-cp'
+  - position: 0
+    prefix: ''
+    separate: false
+    valueFrom: /usr/bin/Marianas-1.8.0.jar
+  - position: 0
+    prefix: ''
+    separate: false
+    valueFrom: org.mskcc.marianas.umi.duplex.fastqprocessing.ProcessLoopUMIFastq
+  - position: 3
+    prefix: ''
```

Comment:
hi Ian, during testing make sure this `prefix: ''` doesn't produce a line with a blank argument, e.g. `'' \`

Reply:
I'm not sure why, but this hasn't caused an issue for me as it did for Ronak / Shalabh

Reply:
remove to be safe